lib/service.dart

```diff
@@ -192,6 +232,243 @@ class IsolateManager {
+  final Map<String, ServiceExtensionState> _enabledServiceExtensions =
+      <String, ServiceExtensionState>{};
+
+  /// Temporarily stores service extensions that we need to add. We should not add
+  /// extensions until the first frame event has been received [firstFrameEventReceived].
+  final Set<String> _pendingServiceExtensions = Set<String>();
+
+  void _handleExtensionEvent(Event event) {
+    final String extensionKind = event.extensionKind;
+    if (event.kind == 'Extension' && extensionKind == 'Flutter.FirstFrame' ||
+        extensionKind == 'Flutter.Frame') {
+      _onFrameEventReceived();
+    }
+  }
+
+  void _onFrameEventReceived() {
+    if (firstFrameEventReceived) {
+      // The first frame event was already received.
+      return;
+    }
+    firstFrameEventReceived = true;
+
+    _pendingServiceExtensions.forEach(_addServiceExtension);
+    _pendingServiceExtensions.clear();
+  }
+
+  void _addRegisteredExtensionRPCs(IsolateRef isolateRef) async {
+    if (_service == null) {
+      return;
+    }
+    final Isolate isolate = await _service.getIsolate(isolateRef.id);
+    if (isolate.extensionRPCs != null) {
+      isolate.extensionRPCs.forEach(_maybeAddServiceExtension);
+
+      if (!firstFrameEventReceived) {
+        final EvalOnDartLibrary flutterLibrary = new EvalOnDartLibrary(
+          'package:flutter/src/widgets/binding.dart',
+          _service,
+        );
+        flutterLibrary
+            .eval('WidgetsBinding.instance.debugDidSendFirstFrameEvent')
+            .future
+            .then((InstanceRef value) {
+          final bool didSendFirstFrameEvent = value.valueAsString == 'true';
+          if (didSendFirstFrameEvent) {
+            _onFrameEventReceived();
+          }
+        });
+      }
+    }
+  }
+
+  void _maybeAddServiceExtension(String name) {
+    if (firstFrameEventReceived) {
+      assert(_pendingServiceExtensions.isEmpty);
+      _addServiceExtension(name);
+    } else {
+      _pendingServiceExtensions.add(name);
+    }
+  }
+
+  void _addServiceExtension(String name) {
+    final StreamController<bool> streamController =
+        _getServiceExtensionController(name);
+
+    _serviceExtensions.add(name);
+    streamController.add(true);
+
+    // Restore any previously enabled states by calling their service extensions.
+    if (_enabledServiceExtensions.containsKey(name)) {
+      _callServiceExtension(name, _enabledServiceExtensions[name].value);
+    }
+  }
+
+  void _callServiceExtension(String name, dynamic value) {
+    if (_service == null) {
+      return;
+    }
+
+    if (value is bool) {
+      _service.callServiceExtension(
+        name,
+        isolateId: _isolateManager.selectedIsolate.id,
+        args: <String, bool>{'enabled': value},
+      );
+    } else if (value is String) {
+      _service.callServiceExtension(
+        name,
+        isolateId: _isolateManager.selectedIsolate.id,
+        args: <String, String>{'value': value},
```

Comment:
omit the type for the args map. It will be inferred to what you wrote and the type doesn't matter anyway for this case.

Reply:
Removing the type gives the error "Missing type arguments for map literal."

Reply:
Commented out strong-mode: implicit-dynamic to prevent this error.